extension/user/login.js

```diff
@@ -0,0 +1,53 @@
+const crypto = require('crypto')
+const {EINVALIDCREDENTIALS, ENOTFOUND, customError} = require('../error')
+/**
+ * @typedef {Object} LoginInputArgs
+ * @property {string} strategy
+ * @property {Object} parameters
+ * @property {string} parameters.login
+ * @property {string} parameters.password
+ */
+
+/**
+ * @param {SDKContext} context
+ * @param {LoginInputArgs} input
+ * @param {function} cb
+ */
+module.exports = function (context, input, cb) {
+  // is logged in
+  if (context.meta.userId) {
+    return cb(null, {
+      userId: context.meta.userId
+    })
+  }
+
+  context.storage.extension.get(input.parameters.login, (errUserId, userId) => {
+    if (errUserId) {
+      return cb(errUserId)
+    }
+
+    if (!userId) {
+      return cb(customError(ENOTFOUND, 'User is not found'))
+    }
+
+    context.storage.extension.get(userId, (err, user) => {
+      if (err) {
+        return cb(err)
+      }
+
+      if (!user) {
+        return cb(customError(ENOTFOUND, 'User not found'))
+      }
+
+      // Check password match
+      const password = crypto.createHash('md5').update(input.parameters.password).digest('hex')
+      if (user.password !== password) {
+        return cb(customError(EINVALIDCREDENTIALS, 'Password mismatch'))
```

Comment:
Same as before: Username and password should always be returned in one single context and the resulting message needs to be exactly the same as above, so no one can see if the email address was even registered.